README.rst

```diff
@@ -258,46 +258,62 @@ For example:
 How To Upgrade Schema
 =====================
 
+We are using Alembic framework to facility schema migration. For more details,
+see https://alembic.sqlalchemy.org/en/latest/tutorial.html.
+
 Here's an example of a normal workflow for performing schema update.
 
-1. create a new changeset file in *pint_server/pint_db_migrate/versions/*. The
-   new changeset file must have the following format: *<d><d><d>_<string>.py*.
-   The first three digit of the filename is the version number, follow by an
-   underscore and a meaningful name of the changeset. The new changeset must
-   have the highest version number, which is usually a plus one increment of
-   the last version. For example, if *pint_server/pint_db_migrate/versions/*
-   currently contains a file *001_in_the_beginning.py*. The next changeset
-   should starts with *002*. Say if we want to add a new column *foo* to the
-   *amazonimages* table. We should create a new file named *002_add_foo_column_to_amazonimages.py* with the following content:
+1. create the Python 3.6 development virtual environment
 
    .. code-block::
 
-     from sqlalchemy import Table, MetaData, String, Column
+     ./bin/create_dev_venv.sh
 
-     def upgrade(migrate_engine):
-         meta = MetaData(bind=migrate_engine)
-         amazonimages = Table('amazonimages', meta, autoload=True)
-         foo = Column('foo', String(100))
-         foo.create(amazonimages)
+2. activate the development virtual environment
 
-     def downgrade(migrate_engine):
-         meta = MetaData(bind=migrate_engine)
-         amazonimages = Table('amazonimages', meta, autoload=True)
-         amazonimages.c.foo.drop()
+   .. code-block::
+
+     source dev_venv/bin/activate
 
-2. create the Python 3.6 development virtual environment
+3. update `pint_server/models.py` to reflect the latest changes
+
+4. if schema change is pretty vanilla and straight forward (i.e.
+   add a new table and add/remove non-indexing columns) you can just let
+   Alembic generate the change script for you. Otherwise, skip to the next
+   step. To do that, you first need to edit `pint_server/alembic.ini` file to
+   update the `sqlalchemy.url` property to point to your DB instance since
+   Alembic will use the existing DB as the baseline. Then use the
+   `--autogenerate` option to auto generate the change script. For example:
 
    .. code-block::
 
-     ./bin/create_dev_venv.sh
+   cd public-cloud-info-service/pint_server
+   alembic revision --autogenerate -m 'add some table'
+
+   If the above command is successful, you'll see the auto generate
+   revision file in `./pint_db_migrate/versions/`. The file is named
+   `<revision>_add_some_table.py`.
 
-3. activate the development virtual environment
+   **NOTE:** don't forget to back out your changes to `pint_server/alembic.ini`
```

Comment:
See above and below comments about providing a reference alembic.ini.exaple that needs to be copied and customised; that represents a better approach in the long run for avoidiung risk of people committing their creds in a PR...